File: src/cpp_package/src/number_counter.cpp
#include "cpp_package/number_counter.hpp"

NumberCounter::NumberCounter() : Node("number_counter"), sum_(0)
{
    subscriber_ = this->create_subscription<example_interfaces::msg::Int64>("number", 1, std::bind(&NumberCounter::ReadNumber, this, std::placeholders::_1));
    publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number_count", 1);
}

void NumberCounter::ReadNumber(const example_interfaces::msg::Int64::SharedPtr msg )
{
    RCLCPP_INFO(this->get_logger(), "Received number %ld: ", msg->data);
    sum_ += msg->data;
    auto sum_msg = example_interfaces::msg::Int64();
    sum_msg.data = sum_;
    publisher_->publish(sum_msg);
}

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<NumberCounter>();

    rclcpp::spin(node);

    rclcpp::shutdown();
    return 0;
}

File: simple_ros_components/src/py_package/py_package/number_counter.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64 

class NumberCounter(Node):
    def __init__(self):
        super().__init__("number_counter")
        self.subscriber_ = self.create_subscription(Int64, "number", self.callback_number_publish, 1)
        self.publisher_ = self.create_publisher(Int64, "number_count", 1)
        self.sum_ = 0
        self.get_logger().info("Number counter is ready: ")


    def callback_number_publish(self, msg:Int64):
        self.get_logger().info(f"Received number:{msg.data}")
        self.sum_ += msg.data
        new_msg = Int64()
        new_msg.data = self.sum_
        self.publisher_.publish(new_msg)

def main(args=None):
    rclpy.init(args=args)

    node = NumberCounter()

    rclpy.spin(node)

    rclpy.shutdown()


if __name__ == "__main__":
    main()    